app/jobs/csv_import_job.rb

```diff
@@ -0,0 +1,112 @@
+require 'csv_parser'
+class CsvImportJob < ActiveJob::Base
+  queue_as :ingest
+
+  before_enqueue do |job|
+    log = job.arguments.last
+    log.pending_job(self)
+  end
+
+  # This copies metadata from the passed in attribute to all of the works that
+  # are members of the given upload set
+  # @param [User] user
+  # @param [Array<Sufia::UploadedFile>] uploaded_files
+  # @param [Hash] attributes attributes to apply to all works
+  # @param [BatchCreateOperation] log
+  def perform(user, csv_source, uploaded_files, selected_files, attributes, log)
+    log.performing!
+
+    create(user, csv_source, uploaded_files, selected_files, attributes, log)
+  end
+
+  private
+
+    def create(user, csv_source, uploaded_files, selected_files, attributes, log)
+
+      uploaded_files_map = {}
+      remote_files_map = {}
+
+      uploaded_files.each do |id|
+        f = Sufia::UploadedFile.find(id).file if !id.blank?
+        uploaded_files_map[File.basename(f.current_path)] = id
+      end
+
+      selected_files.each do |f|
+        remote_files_map[f[:file_name]] = f
+      end
+
+      model = model_to_create(attributes) #::ObjectResource #
+      count = 0
```

Comment:
what is the purpose of `count`? I only see it being incremented at the end of the each block, but not used.